element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/TemplatePropertiesUtil.java

```diff
@@ -109,7 +110,8 @@ public static List<PropertyBuilder> extractTemplatePropertiesFromType(Class<?> t
                 .peek(
                     builder -> {
                       if (nestedPropertiesAnnotation != null
-                          && nestedPropertiesAnnotation.condition() != null) {
+                          && StringUtils.isNotBlank(
+                              nestedPropertiesAnnotation.condition().property())) {
```

Comment:
Hi @chillleader 

The reason for this change is the following:

1. `nestedPropertiesAnnotation.condition()` cannot be `null` as annotation requires constant.
2. Previous change generated the following empty condition.

```
"condition" : {
      "property" : "",
      "oneOf" : [ ],
      "type" : "simple"
    },
```

I wasn't sure that this wasn't expected so it would be nice if you take a look at this snippet too.

Thank you.

Reply:
Makes sense!